src/main/java/com/lovetropics/minigames/common/command/minigames/CommandRegisterMinigame.java

```diff
@@ -1,20 +1,42 @@
 package com.lovetropics.minigames.common.command.minigames;
 
 import com.lovetropics.minigames.common.minigames.MinigameManager;
+import com.lovetropics.minigames.common.minigames.PlayerRole;
 import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
 import net.minecraft.command.CommandSource;
 import net.minecraft.entity.player.ServerPlayerEntity;
 
+import javax.annotation.Nullable;
+
 import static net.minecraft.command.Commands.literal;
 
 public class CommandRegisterMinigame {
 	public static void register(final CommandDispatcher<CommandSource> dispatcher) {
+		// @formatter:off
 		dispatcher.register(
 			literal("minigame")
-			.then(literal("register")
-			.requires(s -> s.getEntity() instanceof ServerPlayerEntity)
-			.executes(c -> CommandMinigame.executeMinigameAction(() ->
-					MinigameManager.getInstance().registerFor((ServerPlayerEntity) c.getSource().getEntity()), c.getSource())))
+				.then(literal("register")
+					.executes(ctx -> registerAsRole(ctx, null))
+					.then(literal("participant")
```

Comment:
Instead of being literals this could read an enum argument and parse the PlayerRole dynamically

Reply:
Hmm, it's not a huge issue, but this would force the arguments to match the actual java enum name, right?

Reply:
I believe you can implement `IStringSerializable` to customize the string representation